climpred/prediction.py

```diff
@@ -83,28 +94,20 @@ def compute_perfect_model(ds,
     """
     supervector_dim = 'svd'
     comparison = get_comparison_function(comparison)
-    if comparison not in [_m2m, _m2c, _m2e, _e2c]:
-        raise ValueError('specify comparison argument')
-
+    _validate_PM_comparison(comparison)
     metric = get_metric_function(metric)
-    if metric in [_pearson_r, _rmse, _mse, _mae]:
-        forecast, reference = comparison(ds, supervector_dim)
-        res = metric(forecast, reference, dim=supervector_dim)
-    # perfect-model only metrics
-    elif metric in [_nmae, _nrmse, _nmse, _ppp, _uacc]:
-        res = metric(ds, control, comparison, running, reference_period)
-    else:
-        raise ValueError('specify metric argument')
+    _validate_PM_metric(metric)
+
+    forecast, reference = comparison(ds, supervector_dim)
+
+    res = metric(forecast,
+                 reference,
+                 dim=supervector_dim,
+                 comparison=comparison)
     return res
 
 
-check_xarray([0, 1])
-def compute_reference(ds,
-                      reference,
-                      metric='pearson_r',
-                      comparison='e2r',
-                      nlags=None,
-                      return_p=False):
```

Comment:
Also why get rid of this? I think we should give the option to return p-values for ACCs (pearson r). The user should be encouraged to check statistical significance.

Reply:
instead of `return_p` you could just use `metric=pearson_r_p_value` in `compute_x` and this does the job. also the number of outputs is now strictly one.
I didnt want to destroy your significance approach, I just think its incorporated anyways via metric.

Reply:
You're right, I didn't even think about that. I might add an alias so the user doesn't need to put in such a long keyword.